chunky/src/java/se/llbit/chunky/model/WaterModel.java

```diff
@@ -73,7 +73,7 @@
   static final double height[] =
       {14 / 16., 12.25 / 16., 10.5 / 16, 8.75 / 16, 7. / 16, 5.25 / 16, 3.5 / 16, 1.75 / 16};
 
-  private static final float[][][] normalMap;
+  private static final float[] normalMap;
```

Comment:
The memory savings are by not using nested arrays?

Reply:
Yes, i explained it on discord but i restate here as it offers more visibility.
Every Java object has a header of a given size that is used by the jvm to do its stuff. The exact size of this header depends on the jvm but for example sake let's say it is 16 bytes (even it the size is different, the idea stays the same).
Arrays are object and have this header. In addition arrays store their length which is an additional 4 bytes that us not needed when the size is a constant. (And for arrays of `long`, `double`, or objects, we can add another 4 bytes so array data is aligned on a multiple of 8 bytes)
For example this variable (when loading a normal map of 512*512) uses 24 + 512 * (24 + 512 * (20 + 2*4)) bytes (7,352,344 bytes).
By using a flat array, the new size is 24 + 512*512*2*4 bytes (2,097,152 bytes), saving a lot of memory.